lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -77,36 +77,29 @@ function dosomething_rogue_send_reportback_to_rogue($values, $user = NULL) {
         'crop_rotate' => $values['crop_rotate'],
   ];
 
-  $response = $client->postReportback($data);
+  try {
+    $response = $client->post($client->getBaseUri() . '/reportback', $data);
 
-  if (in_array($response->code, [200, 201]) && module_exists('stathat')) {
-    stathat_send_ez_count('drupal - Rogue - reportback sent - count', 1);
-  }
-  else {
-    // Log as watchdog error, and stathat value.
     if (module_exists('stathat')) {
-      stathat_send_ez_count('drupal - Rogue - reportback failed - count', 1);
+      stathat_send_ez_count('drupal - Rogue - reportback sent - count', 1);
     }
-    // Save fail to a db log so we can easily export.
-    db_insert('dosomething_rogue_failed_task_log')
-      ->fields([
-        'drupal_id' => $user->uid,
-        'campaign_id' => $values['nid'],
-        'campaign_run_id' => $run->nid,
-        'quantity' => $values['quantity'],
-        'why_participated' => $values['why_participated'],
-        'file' => isset($values['file']) ? $values['file'] : NULL,
-        'caption' => isset($values['caption']) ? $values['caption'] : NULL,
-        'timestamp' => REQUEST_TIME,
-        'response_code' => $response->code,
-        'response_values' => json_encode($response),
-      ])
-      ->execute();
+    if (!$response) {
+      // This is a 404
+      dosomething_rogue_handle_failure($user, $values, $run, $response, $e);
+    }
+  }
+  catch (GuzzleHttp\Exception\ServerException $e) {
+    // These aren't yet caught by Gateway
+    dosomething_rogue_handle_failure($user, $values, $run, $response, $e);
+  }
+  catch (Exception $e) {
```

Comment:
`"DoSomething\Gateway\Exceptions\BadRequestException" 
 "DoSomething\Gateway\Exceptions\ApiException"`

the second one prints out that when it throws `new BadRequestException($endpoint, json_encode($response))`